### ISSUE TITLE
Potential Warning to address when generating transformers

### ISSUE THREAD

**Issue by elegoupil** [Pleias/Pleias-RAG-Library#11]
Testing! 

When loading the model, I get --

```
/usr/local/lib/python3.10/site-packages/transformers/generation/configuration_utils.py:636: UserWarning: `do_sample` is set to `False`. However, `top_p` is set to `0.95` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `top_p`.
  warnings.warn(
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:2 for open-end generation.
The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
```

It looks like the `_generate_transformers` method shall be adjusted to properly handle the attention mask and pad token settings....

**Comment by vicastiel** [comment]
Here’s what fixed it:

Inside the _init_vllm() method of the RAGWithCitations class, after initializing the tokenizer, add this:

Manually ensure special tokens are present
special_tokens = [
    "<|query_start|>", "<|query_end|>", "<|source_start|>", "<|source_id|>", "<|source_end|>",
    "<|language_start|>", "<|language_end|>", "<|query_analysis_start|>", "<|query_analysis_end|>",
    "<|query_report_start|>", "<|query_report_end|>", "<|source_analysis_start|>", "<|source_analysis_end|>",
    "<|source_report_start|>", "<|source_report_end|>", "<|draft_start|>", "<|draft_end|>",
    "<|answer_start|>", "<|answer_end|>"
]
missing = [tok for tok in special_tokens if tok not in tokenizer.additional_special_tokens]
if missing:
    tokenizer.add_special_tokens({"additional_special_tokens": missing})

Then use only <|answer_end|> as the stop token:

stop_token_ids = [tokenizer.convert_tokens_to_ids("<|answer_end|>")]

(Optional) Patch output post-generation:
If the model forgets to include <|answer_end|>, you can add it manually:
if "<|answer_end|>" not in raw_response:
    raw_response += "\\n<|answer_end|>"

About this warning:

`do_sample` is set to False, but `top_p` is set to 0.95 — this flag is only used for sampling

This is normal — if you’re using temperature=0.0, it means greedy decoding, and top_p is simply ignored. Nothing to worry about.

Hope this helps!
